### PR TITLE
azure - unblock prototyping event based functions

### PR DESCRIPTION
**Pull request by stefangordon**
We'll end up using *authenticated* webhooks for alerts from Azure Monitor for registering Functions for any sort of event (metrics, activity logs, etc).

This is just a few small things to unblock - with these changes I was able to deploy a function that would accept a  manually configured Monitor Alert Action.

This also starts adding in the status check re: #2592

### CODE REVIEW

File: tools/c7n_azure/c7n_azure/policy.py [hunk 1]
```diff
@@ -73,7 +74,11 @@ def provision(self):
 
         archive = FunctionPackage(self.policy.data)
         archive.build()
-        archive.publish(parameters['name']['value'])
+
+        if archive.status(parameters['name']['value']):
```
Comment (aluong) [inline tools/c7n_azure/c7n_azure/policy.py]: Let's move `parameters['name']['value']` to a local variable.

File: tools/c7n_azure/c7n_azure/function_package.py [hunk 2]
```diff
@@ -99,11 +99,14 @@ def _add_function_config(self):
             binding['schedule'] = self.policy['mode']['schedule']
 
         elif mode_type == 'azure-stream':
```
Comment (aluong) [inline tools/c7n_azure/c7n_azure/function_package.py]: Do we want to keep this name? I guess it doesn't matter what the underlying trigger is if the semantics of the mode makes sense.

Comment (stefangordon) [reply tools/c7n_azure/c7n_azure/function_package.py]: probably not - so much of this is going to get a rewrite I'm mostly just trying to unblock and we can finalize naming once we see what it does.

File: tools/c7n_azure/c7n_azure/policy.py (hunk 1 above)
Comment (stefangordon) [reply tools/c7n_azure/c7n_azure/policy.py]: Oh it was one, just wasn't being used :)

**Pull request merged**